components/clarinet-cli/src/frontend/cli.rs

```diff
@@ -1262,6 +1268,69 @@ pub fn main() {
         Command::Integrate(cmd) => {
             let manifest = load_manifest_or_exit(cmd.manifest_path);
             println!("Computing deployment plan");
+
+            let mut orchestrator = match DevnetOrchestrator::new(manifest.clone(), None) {
+                Ok(orchestrator) => orchestrator,
+                Err(e) => {
+                    println!("{}", format_err!(e));
+                    process::exit(1);
+                }
+            };
+
+            let cache_location =
+                if let FileLocation::FileSystem { path } = &manifest.project.cache_location {
+                    path.clone()
+                } else {
+                    println!("cache location must be a file system path");
+                    process::exit(1);
+                };
+
+            let devnet_config = orchestrator
+                .network_config
+                .as_ref()
+                .and_then(|c| c.devnet.as_ref())
+                .expect("devnet configuration not found");
+
+            let _ = fs::create_dir(cache_location.clone());
+            let _ = fs::create_dir(devnet_config.working_dir.clone());
+            let _ = fs::create_dir(format!("{}/conf", devnet_config.working_dir));
+            let _ = fs::create_dir(format!("{}/data", devnet_config.working_dir));
+
+            if devnet_config.enable_subnet_node {
+                let subnet_deployer =
+                    match QualifiedContractIdentifier::parse(&devnet_config.subnet_contract_id) {
+                        Ok(contract) => contract,
+                        Err(e) => {
+                            println!("invalid subnet contract id: {}", e);
+                            process::exit(1);
+                        }
+                    }
+                    .issuer;
+                let subnet_leader = compute_stx_address(
+                    &devnet_config.subnet_leader_mnemonic,
+                    &devnet_config.subnet_leader_derivation_path,
+                    &StacksNetwork::Devnet,
+                );
+                let _ = fs::create_dir(format!("{}/requirements", cache_location.display()));
+
+                let ctx = Context {
+                    logger: None,
+                    tracer: false,
+                };
+                match block_on(orchestrator.prepare_subnet_node_container(
+                    1,
+                    &ctx,
+                    cache_location,
+                    &subnet_deployer,
+                    &subnet_leader.into(),
+                )) {
+                    Ok(_) => {}
+                    Err(e) => {
+                        process::exit(1);
+                    }
+                };
+            }
+
```

Comment:
This all looks fine to me, but it seems like a lot of code to add to a CLI command branch -- can this be easily refactored into a function?

Reply:
You're right. It would be nice if each of these cases were just a function call. For now, I could refactor this new code into `setup_directories` and `setup_subnet` or something along those lines... although, the initialization of `orchestrator`, `cache_location`, and `devnet_config` would make sense to all stay at the top-level, so it wouldn't really reduce it by much. How about we leave this refactoring of all of the commands for later?

Reply:
Works for me